app/src/main/java/com/github/polypoly/app/ui/menu/lobby/GameLobbyActivity.kt

```diff
@@ -70,162 +83,623 @@ class GameLobbyActivity : ComponentActivity() {
+
+    /**
+     * Display the current player count in relation to the maximum player count.
+     * @param players the list of players in the game lobby.
+     * @param maxPlayers the maximum number of players allowed in the game lobby.
+     */
+    @Composable
+    fun CurrentPlayerCountDisplay(players: List<User>, maxPlayers: Int) {
+        Row {
+            Text(
+                text = getString(R.string.game_lobby_players_title),
+                style = MaterialTheme.typography.body1,
+                modifier = Modifier
+                    .testTag("game_lobby_start_game_button_current_player_count_title")
+            )
+            Text(
+                text =  getString(R.string.game_lobby_players_count, players.size.toString(), maxPlayers.toString()),
+                style = MaterialTheme.typography.body1.copy(
+                    color = MaterialTheme.colors.primary
+                ),
+                modifier = Modifier
+                    .testTag("game_lobby_start_game_button_current_player_count")
+            )
+        }
+    }
+
+    /**
+     * Launch the game activity and finish this activity. Only available to the admin.
+     * will set the game lobby as started in the database.
+     */
     private fun launchGameActivity(packageContext: Context) {
         val completedLobby = gameLobbyWaitingModel.getGameLobby().value
         if (completedLobby != null) {
-            GameRepository.game = Game.launchFromPendingGame(completedLobby)
-            val gameIntent = Intent(packageContext, GameActivity::class.java)
-            startActivity(gameIntent)
-            finish()
+
+            val newGameLobby = GameLobby(
+                completedLobby.admin,
+                completedLobby.rules,
+                completedLobby.name,
+                completedLobby.code,
+                completedLobby.private,
+                true
+            )
+            remoteDB.updateValue(newGameLobby)
+
+            navigateToGame(packageContext, newGameLobby)
         }
     }
 
-    @Composable
-    private fun QuitButton() {
-        Button(onClick = { /*TODO*/ }) {
-            Text(text = "QUIT")
+    /**
+     * Leave the lobby and return to the main menu.
+     * Makes sure that the admin is replaced by another user if the admin leaves.
+     */
+    private fun leaveLobby(gameLobby: GameLobby) {
+        val newUsersRegistered = gameLobby.usersRegistered.filter {it.id != currentUser.id}
+        if (newUsersRegistered.isEmpty()){
+            remoteDB.removeValue<GameLobby>(gameLobby.code)
+        } else {
+            val newAdmin =
+                if (currentUser.id == gameLobby.admin.id && !gameLobby.started) newUsersRegistered.random() else gameLobby.admin
+            val newGameLobby = GameLobby(
+                newAdmin,
+                gameLobby.rules,
+                gameLobby.name,
+                gameLobby.code,
+                gameLobby.private,
+                gameLobby.started
+            )
```

Comment:
`GameLobby ` is a data class so you can do
```kotlin
val newGameLobby = gameLobby.copy(admin = newAdmin)
```

Reply:
looks great thanks!